miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java

```diff
@@ -298,10 +306,12 @@ public ModelAndView editPools(@RequestParam("ids") String poolIds, ModelMap mode
   private static class BulkMergePoolsBackend extends BulkTableBackend<PoolDto> {
 
     private final PoolService poolService;
+    private final BoxDto newBox;
 
-    public BulkMergePoolsBackend(PoolService poolService) {
+    public BulkMergePoolsBackend(BoxDto newBox, PoolService poolService) {
       super("pool", PoolDto.class);
       this.poolService = poolService;
+      this.newBox = newBox;
     }
 
     protected PoolDto createDtoFromParents(List<Long> parentIds, List<Integer> proportions) throws IOException {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 21 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)